src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -68,6 +70,28 @@ public static class TagHelperBlockRewriter
                 // Only want to track the attribute if we succeeded in parsing its corresponding Block/Span.
                 if (succeeded)
                 {
+                    string attributeValueType;
+
+                    // Check if it's a bound attribute that is not of type string. If there's an empty value we need 
+                    // to create an error.
+                    if (attributeValueTypes.TryGetValue(attribute.Key, out attributeValueType) &&
+                       !IsStringAttribute(attributeValueType) &&
+                       (attribute.Value == null ||
+                       (!attribute.Value.IsBlock && string.IsNullOrEmpty(((Span)attribute.Value).Content)) ||
+                       (attribute.Value.IsBlock && ((Block)attribute.Value).Children.Count() == 0)))
```

Comment:
No, being a block simply means there's the potential to have more than 1 part of the value.